platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

```diff
@@ -47,6 +47,12 @@ In previous versions of Gradle, toolchain provisioning could leave a partially p
 
 The embedded Kotlin has been updated from 1.9.22 to link:https://github.com/JetBrains/kotlin/releases/tag/v1.9.23[Kotlin 1.9.23].
 
+==== Compiling against Gradle implementation classpath
+
+In previous versions of Gradle, some JVM projects that did not declare any dependencies were erroneously compiled against the Gradle API.
+This meant that compilations without any declared dependencies were able to compile against Gradle classes without declaring a `gradleApi` dependency.
+Declare a `gradleApi` dependency or use the `java-gradle-plugin` plugin if you wish to compile against the Gradle API.
+
```

Comment:
Some rewording suggestions...

I read between the lines here, but I think if you had a project like this, you wouldn't be able to really do much with it.  The IDE wouldn't know about these dependencies and you couldn't have tests either. 

I think the actual classpath is roughly what the daemon needs to startup and not the entire Gradle runtime or API.

```suggestion
==== Implicitly compiling against Gradle's runtime

In previous versions of Gradle, it was possible for Java projects that had no declared dependencies to implicitly compile against Gradle's runtime classes.

This means it was possible for some projects to compile without any declared dependencies, but we think this is very unlikely to be encountered with real projects because other things were unlikely to work (IDE integration and test execution).

If you need to use the Gradle API, declare a dependency on `gradleApi()` or use the `java-gradle-plugin` plugin.
```